runtime/pkg/formatter/intervals.go

```diff
@@ -0,0 +1,219 @@
+package formatter
+
+import (
+	"fmt"
+	"strings"
+
+	"golang.org/x/text/language"
+	"golang.org/x/text/message"
+	xnumber "golang.org/x/text/number"
+)
+
+// Constants for time calculations
+const (
+	msPerSec   = 1000
+	msPerMin   = 60 * msPerSec
+	msPerHour  = 60 * msPerMin
+	msPerDay   = 24 * msPerHour
+	msPerMonth = 30 * msPerDay
+	msPerYear  = 365 * msPerDay
+)
+
+type timeunit string
+
+const (
+	tuMs    timeunit = "ms"
+	tuSec   timeunit = "s"
+	tuMin   timeunit = "m"
+	tuHour  timeunit = "h"
+	tuDay   timeunit = "d"
+	tuMonth timeunit = "mon"
+	tuYear  timeunit = "y"
+)
+
+type intervalFormatter struct{}
+
+func (f *intervalFormatter) stringFormat(x any) (string, error) {
+	if v, ok := asNumber[float64](x); ok {
+		return formatMsInterval(v), nil
+	}
+	return "", fmt.Errorf("not a number: %v", x)
+}
+
+/**
+ * Formats a millisecond value into a compact human-readable time interval.
+ *
+ * The strategy is to:
+ * - show two digits of precision
+ * - prefer to show two integer digits in a smaller unit
+ * - if that is not possible, show a floating point number in a larger unit with one digit of precision (e.g. 1.2 days)
+ *
+ * see https://www.notion.so/rilldata/Support-display-of-intervals-and-formatting-of-intervals-e-g-25-days-in-dashboardsal-data-t-8720522eded648f58f35421ebc28ee2f
+ */
+
+func formatMsInterval(ms float64) string {
```

Comment:
Use the usual Go-style comments (`//` on each line). And avoid an empty line before the function definition (it decouples the docstring so the editor hints disappear). Applies a few places.